### PR TITLE
Fix ClipPlane being unset

### PR DESCRIPTION
**Pull request by sebavan**
https://forum.babylonjs.com/t/clipping-plane-potential-bug/37851

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/13497/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13497/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13497/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13497/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13497/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13497/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13497/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13497/merge#BCU1XR#0

File: packages/dev/core/src/Materials/clipPlaneMaterialHelper.ts [hunk 1]
```diff
@@ -65,19 +65,24 @@ function setClipPlane(effect: Effect, uniformName: string, clipPlane: Nullable<P
...
-            if (!isSet) {
-                defines.push(defineString);
-            }
-        } else {
-            isSet = defines[defineName];
-            defines[defineName] = true;
+function addDefine(clipPlane: Nullable<Plane>, defines: string[] | Record<string, any>, defineName: string, defineString: string): boolean {
+    const defineClipPlane = clipPlane ? true : false;
+
+    let aleradySet: boolean;
```
Comment (Popov72) [inline packages/dev/core/src/Materials/clipPlaneMaterialHelper.ts:71]: neat: should be `alreadySet` :)

**Review by Popov72 — APPROVED** [review]

File: packages/dev/core/src/Materials/clipPlaneMaterialHelper.ts (hunk 1 above)
Comment (sebavan) [reply packages/dev/core/src/Materials/clipPlaneMaterialHelper.ts:71]: YUP :-)

**Pull request merged**